### PR TITLE
[567909] Improve analysis selector to be more user friendly

### PR DESCRIPTION
**Pull request by SteveMonnier**
There as been some improvement on the analysis selector to be more user
friendly. As the one provided in Capella was better that the one in
Sirius, it was made as a class (instead of a nested class) so it could
be reused properly.
Furthermore, the move representation actions have been modified. Instead
of having an action for each available aird file, there is now only one
action that displays the analysis selector.

Bug:567909
Change-Id: I8398171709a09548a863fd8edd1d163addc2b3a6
Signed-off-by: Steve Monnier <steve.monnier@obeo.fr>

### CODE REVIEW

**Comment by SteveMonnier** [comment]
This commit depends on https://bugs.eclipse.org/bugs/show_bug.cgi?id=567805

**Comment by lfasani** [comment]
java.util.NoSuchElementException when clicking Move on a Representation in a non-fragmented local Capella project.
The Move menu entry should not be visible.

**Comment by SteveMonnier** [comment]
> java.util.NoSuchElementException when clicking Move on a Representation in a non-fragmented local Capella project.
> The Move menu entry should not be visible.

Oh right, on Sirius I don't have the action if there is only one DAnalysis, but I did not report that on Capella's action provider. I am going to fix it.

File: core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * 
+ * @author <a href="mailto:steve.monnier@obeo.fr">Steve Monnier</a>
+ */
+public class MoveRepresentationsAction extends BaseSelectionListenerAction {
+
+    /**
+     * Constructs the action allowing to open representations.
+     */
+    public MoveRepresentationsAction() {
+        super("Move"); //$NON-NLS-1$
```
[the remark this answers is not in the document]
Comment (SteveMonnier) [reply core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java:38]: Are you sure? I took exemple on OpenRepresentationsAction in the same package and it has "super("Open"); //$NON-NLS-1$"

File: core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import org.eclipse.sirius.viewpoint.ViewpointPackage;
+import org.eclipse.ui.actions.BaseSelectionListenerAction;
+import org.polarsys.capella.core.model.handler.helpers.RepresentationHelper;
+
+/**
+ * The action allowing to move representations to a different {@link DAnalysis}.
+ * 
+ * @author <a href="mailto:steve.monnier@obeo.fr">Steve Monnier</a>
+ */
+public class MoveRepresentationsAction extends BaseSelectionListenerAction {
```
[the remark this answers is not in the document]
Comment (SteveMonnier) [reply core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java:32]: Fixed

File: core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+     */
+    @Override
+    public void run() {
+        Collection<DRepresentationDescriptor> repDescriptors = getValidRepresentationDescriptors(getStructuredSelection());
+
+        // Precondition
+        if (repDescriptors.isEmpty()) {
+            return;
+        }
+        DRepresentationDescriptor next = repDescriptors.iterator().next();
```
[the remark this answers is not in the document]
Comment (SteveMonnier) [reply core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java:70]: Ah, I took the different approach that the user is asked where to place the representation for each selected representation. I will wait for confirmation to change it.

File: core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (pdulth) [reply core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java:38]: If this message is displayed, would be better yes.

File: core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
+/*******************************************************************************
```
Comment (pdulth) [inline core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java:1]: EPL2 Licence

Comment (SteveMonnier) [reply core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java:1]: Done

File: core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (SteveMonnier) [reply core/plugins/org.polarsys.capella.core.sirius.ui/src/org/polarsys/capella/core/sirius/ui/actions/MoveRepresentationsAction.java:41]: Done

**Comment by pdulth** [comment]
Merged by #419

**Pull request closed without merging**